app/models/druid_version_zip.rb

```diff
@@ -0,0 +1,35 @@
+# Just a regular model, not an ActiveRecord-backed model
+class DruidVersionZip
+  attr_reader :druid, :version
+
+  # @param [String] druid
+  # @param [Integer] version
+  def initialize(druid, version)
+    @druid = DruidTools::Druid.new(druid.downcase)
+    @version = version
+  end
+
+  # @return [String]
+  # @see [S3 key name performance implications] https://docs.aws.amazon.com/AmazonS3/latest/dev/request-rate-perf-considerations.html
+  # @example return 'ab/123/cd/4567/ab123cd4567/ab123cd4567.v0001.zip'
+  def s3_key
+    druid.tree.join('/') + format(".v%04d.zip", version)
+  end
+
+  # @return [File] opened zip file
+  def file
+    File.open(file_path)
+  end
+
+  # @return [String] Path to the local temporary transfer zip
+  def file_path
```

Comment:
I did some additional reading on Pathname and found this:
```


    Pathname represents a pathname which locates a file in a filesystem. The pathname depends on OS: Unix, Windows, etc. Pathname library works with pathnames of local OS. However non-Unix pathnames are supported experimentally.

    It does not represent the file itself. A Pathname can be relative or absolute. It’s not until you try to reference the file that it even matters whether the file exists or not.

    Pathname is immutable. It has no method for destructive update.
```

which seems like overkill here.  So file_path being a method that returns a string ... I can live with that.  :-)